src/components/ui/image.jsx

```diff
@@ -30,6 +30,15 @@ let Image = React.createClass({
     }
   },
 
+  componentWillReceiveProps(nextProps) {
+    let { src } = this.state
+
+    this.setState({
+      isLoaded: nextProps.src !== src,
+      src: nextProps.src
+    });
+  },
```

Comment:
Just a stray thought. Do we have test coverage on this file? If not, this might be a good thing to cover.